components/track/TrackMenu.vue

```diff
@@ -47,6 +51,19 @@ const dropdownMenuItemsForTrack = (track: TrackModel) => {
     });
   }
 
+  if (runtimeConfig.public.systemName !== "Electron") {
+    items.push({
+      icon: "icon.download",
+      text: t("track.dropdown.download"),
+      clickFunction: async () => {
+        const result = await download(track);
+        if (result === "no-permission") {
```

Comment:
is there a better way than having "magic strings"?
For example by splitting it into to parts:
- hasPermissionToDownload()
- download()

Another way would be to introduce a strongly typed return value

Reply:
is **is** a strongly typed return value: DownloadResult.